components/bl-carousel-component/src/index.js

```diff
@@ -0,0 +1,138 @@
+import { useState, useEffect } from 'react';
+import { NextButton, PrevButton, ListItem, CarouselIndicators } from './subcomponents';
+import { getPrevImg, getNextImg } from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function Carousel({ component, eventHandlers }) {
+  const {
+    display,
+    classList,
+    imagesData,
+    heightImage,
+    hasAutoplay,
+    autoplayDelay,
+    animationDuration,
+    isControlButtonsVisible,
+    isIndicatorsVisible,
+    typeAnimation
+  } = component;
+  const { onNextButton, onPrevButton, onHover, onUnhover } = eventHandlers;
+
+  const [data, setData] = useState(imagesData);
+  const [currentImg, setCurrentImg] = useState(0);
+  const [isNextAnimation, setIsNextAnimation] = useState(false);
+  const [isPrevAnimation, setIsPrevAnimation] = useState(false);
```

Comment:
If I understand correctly, the problem is if currentImage is the last image and the user clicks Next, the animation will be isPrevImage instead of isNextImage

inNextAnimation
![nextAnimaion](https://user-images.githubusercontent.com/99363993/185607428-991b40ee-b460-4216-8d88-7690aedfda3f.gif)
isPrevAnimation
![prevAnimation](https://user-images.githubusercontent.com/99363993/185607482-9c77651a-95e0-4b1d-bf5d-b32f398e10e2.gif)